connection_pool/connection_pool.go

```diff
@@ -526,6 +526,9 @@ func (connPool *ConnectionPool) EvalAsync(expr string, args interface{}, userMod
 
 // Do sends the request and returns a future.
 func (connPool *ConnectionPool) Do(req tarantool.Request, userMode Mode) *tarantool.Future {
+	if stickyReq, ok := req.(tarantool.StickyRequest); ok {
+		return stickyReq.Conn().Do(req)
```

Comment:
I think we need a validation here that Conn() belongs to the pool. And inside Connection.Do() too.

Reply:
Added.